tag_test.go

```diff
@@ -11,25 +11,25 @@ var parserTests = []struct {
 	errMsg string
 }{
 	{"", nil, "Empty tagname"},
-	{" ", nil, "Whitespace"},
-	{"\t", nil, "Whitespace"},
-	{"\n", nil, "Whitespace"},
-	{"\b", nil, "Non-ASCII"},
+	{" ", nil, "whitespace"},
+	{"\t", nil, "whitespace"},
+	{"\n", nil, "whitespace"},
+	{"\b", nil, "non-alphabetic character"},
```

Comment:
As I think about this more, I'd more likely use a custom `Unwrap` to return `ErrTagString` instead of `%w` to avoid having to fit the "invalid tag string" wording into each message. 
```
const ErrTagString = errors.New("Invalid tag string")

// tagNameErr is a private error type that wraps an ErrTagString
type tagNameErr struct { Msg string }
func (err tagNameErr) Error() string { return err.Msg }
func (err tagNameErr) Unwrap() error { return ErrTagString }
```

But is this worth doing? 🤷‍♀️ Do users of our code ever need to check if the returned error is due to an invalid tag string? Right now we're the only users and we're probably loading our tag strings from known good sources.

Reply:
Given that these tag requests are coming from "the outside world" (ie. the agent program, over a socket) I'm not sure that it makes sense to factor out errors into types, since they'll get marshalled into strings when the response goes over the wire anyway?  What do you think?

Reply:
We might expose these errors via `$URL/raw`, but that's not for long-term use. I expect the gRPC API used by the agents won't ever expose one of these hypothetical `ErrTagString` errors because agents won't ever be providing a tag string. It's possible but very unlikely we'd ever want to check for this error type. It could be useful for some hypothetical third-party user of the `go-plc` code (but those users will likely never exist).

I wouldn't expose multiple error types here; I'd just return `ErrTagString` with different messages.

Regardless, I don't think it's a worthwhile change right now. My above comments were just putting too much thought into a potential future design.

Reply:
That said, now that I think about this, I agree certainly that we should be returning singleton errors rather than ad-hoc ones.